### PR TITLE
More gold data tests

### PR DESCRIPTION
**Pull request by bickelj**
Improve gold data tests with some refactoring and additional examples of changemaker data conflicts.

Issue #1257

### CODE REVIEW

**Comment by bickelj** [comment]
@slifty @jasonaowen What do you think of this approach of pulling some of the more common elements that are not test-specific into a `beforeEach` in a dedicated `describe` block? One thing I like is that the `beforeEach` can have the calls to production code, thus removing those calls from test coverage, thus making the test coverage more honest and separating those failures out. I might even want to do an extra `describe` and `beforeEach` block for individual tests due to this benefit! One drawback is the separation of the setup from the test. It's no longer self-contained. I can't simply go to the test and read it to see everything that's going on. However, as long as the data are carefully crafted and declared in the same file, it should be easy to see what's happening. As far as I can tell this is like what Meszaros calls the "Shared Fixture."

File: src/__tests__/changemakers.int.test.ts [hunk 1]
```diff
@@ -262,29 +272,53 @@ describe('/changemakers', () => {
...
-				// To advance test changemaker IDs beyond 1-2.
-				await insertTestChangemakers();
-				// This should be ID 3, although unfortunately implicitly, and only when tests run in serial.
-				const changemakerId = (
-					await createChangemaker({
-						name: 'Five thousand one hundred forty seven reasons',
-						taxId: '05119',
-					})
-				).id;
+			beforeEach(async () => {
```
Comment (jasonaowen) [inline src/__tests__/changemakers.int.test.ts:292]: This is great! I like this approach; good idea, @bickelj.

Comment (bickelj) [reply src/__tests__/changemakers.int.test.ts:294]: Thanks.

**Comment by bickelj** [comment]
@slifty @jasonaowen My guess that extracting test setup into `beforeEach` makes the test more honest is probably false. I thought I could improve the test coverage honesty with a simple commit that added several `/* istanbul ignore next */` statements before test setup blocks but I saw no change in coverage. Even running a single integration test shows upwards of 50% code coverage across the code base. Because this is a much deeper rabbit hole than I imagined, I created #1267 separately.

**Review by slifty — APPROVED** [review]
Looks good!

As we know over time some of these rules may shift as we get more user data, but this is a great reflection of the expected implementation of our current requirements!

**Comment by bickelj** [comment]
bfa7d7f rebased on main without any other modification.

**Pull request merged**